modules/porous_flow/src/materials/PorousFlowHystereticRelativePermeabilityGas.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowHystereticRelativePermeabilityGas.h"
+#include "PorousFlowVanGenuchten.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowHystereticRelativePermeabilityGas);
+
+InputParameters
+PorousFlowHystereticRelativePermeabilityGas::validParams()
+{
+  InputParameters params = PorousFlowHystereticRelativePermeabilityBase::validParams();
+  params.addRangeCheckedParam<Real>(
+      "gamma", 0.33, "gamma > 0", "Gamma parameter that is used for the gas relative permeability");
+  params.addRangeCheckedParam<Real>(
+      "k_rg_max",
+      1.0,
+      "k_rg_max > 0 & k_rg_max <= 1",
+      "Value of the gas relative permeability at liquid saturation = S_lr");
+  MooseEnum low_ext_enum("linear_like cubic", "linear_like");
+  params.addParam<MooseEnum>(
+      "gas_low_extension_type",
+      low_ext_enum,
+      "Type of extension to use for liquid saturation < S_lr for the gas relative permeability.  "
+      "All extensions employ a cubic whose value is 1.0 at liquid saturation = 0, and whose "
+      "derivative is zero at liquid saturation = 0, and whose value is k_rg_max at liquid "
+      "saturation = S_lr.  linear_like: the derivative at liquid_saturation = S_lr is equal to "
+      "(k_rg_max - 1) / S_lr.  cubic: the derivative at liquid_saturation = S_lr equals the "
+      "derivative of the unextended drying curve at that point");
+  params.addClassDescription(
+      "PorousFlow material that computes relative permeability of the gas phase in 1-phase or "
+      "2-phase models that include hysteresis.  You should ensure that the 'phase' for this "
+      "Material does indeed represent the gas phase");
+  return params;
+}
+
+PorousFlowHystereticRelativePermeabilityGas::PorousFlowHystereticRelativePermeabilityGas(
+    const InputParameters & parameters)
+  : PorousFlowHystereticRelativePermeabilityBase(parameters),
+    _liquid_phase(_phase_num == 0 ? 1 : 0),
```

Comment:
Maybe this prevents you from putting `_liquid_phase` into the base class? If so, ignore that comment

Reply:
yea, that's the reason i can't put `_liquid_phase` in the base class.